src/main/java/ch/njol/skript/expressions/ExprAdvancementMessage.java

```diff
@@ -0,0 +1,108 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.ScriptLoader;
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.log.ErrorQuality;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+import net.kyori.adventure.text.Component;
+
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerAdvancementDoneEvent;
+
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Advancement Message")
+@Description("The advancement message in an advancement completion event.")
+@Examples("set the advancement message to \"%event-player% completed an advancement!\"")
+@RequiredPlugins("Paper 1.16.5+")
+@Since("INSERT VERSION")
+public class ExprAdvancementMessage extends SimpleExpression<String> {
+
+	static {
+		if (Skript.classExists("org.bukkit.event.player.PlayerAdvancementDoneEvent") && Skript.methodExists(org.bukkit.event.player.PlayerAdvancementDoneEvent.class, "message")) {
+				Skript.registerExpression(ExprAdvancementMessage.class, String.class, ExpressionType.SIMPLE, "[the] advancement message");
+		}
+	}
+	
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!getParser().isCurrentEvent(PlayerAdvancementDoneEvent.class)) {
+			Skript.error("Cannot use the advancement message outside an on advancement done event", ErrorQuality.SEMANTIC_ERROR);
```

Comment:
```suggestion
			Skript.error("The 'advancement message' is only usable in advancement done events");
```
for consistency:
https://github.com/SkriptLang/Skript/blob/7f60de10c66e446d456e53c0de0ad933f4579d18/src/main/java/ch/njol/skript/expressions/ExprAbsorbedBlocks.java#L54-L61